core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -383,28 +382,30 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                 final IndexMetaData.Builder tmpImdBuilder = IndexMetaData.builder(request.index());
 
                 final int routingNumShards;
-                if (shrinkFromIndex == null) {
-                    routingNumShards = IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.get(indexSettingsBuilder.build());
+                if (recoverFromIndex == null) {
+                    Settings idxSettings = indexSettingsBuilder.build();
+                    routingNumShards = IndexMetaData.INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING.get(idxSettings);
                 } else {
-                    final IndexMetaData sourceMetaData = currentState.metaData().getIndexSafe(shrinkFromIndex);
+                    final IndexMetaData sourceMetaData = currentState.metaData().getIndexSafe(recoverFromIndex);
```

Comment:
assert that settings doesn't contain INDEX_NUMBER_OF_ROUTING_SHARDS_SETTING ?

Reply:
why is this?

Reply:
because the target index shouldn't have that setting. it get's it from the source.